src/leo-client-app/src/components/EditSchedules.tsx

```diff
@@ -0,0 +1,243 @@
+//   { id: 3, name: 'checkSensor', operator: 'operator1', created: 'Jan 1 2023' },
+//   { id: 4, name: 'loadData', operator: 'operator2', created: 'Jan 2 2023' },
+//   { id: 5, name: 'tearDown', operator: 'operator1', created: 'Jan 1 2023' },
+//   // ... more commands
+// ];
+
+// const EditSchedulePage: React.FC = () => {
+//   return (
+//     <div className="edit-schedule-page">
+//       <div className="header">
+//         <button className="back-button">{"<"}</button>
+//         <h2>Edit Schedule</h2>
+//         <button className="close-button">{"X"}</button>
+//       </div>
+//       <h3>Jan 09 Schedule</h3>
+//       <table>
+//         <thead>
+//           <tr>
+//             <th>id</th>
+//             <th>Command</th>
+//             <th>Operator</th>
+//             <th>Created</th>
+//           </tr>
+//         </thead>
+//         <tbody>
+//           {commands.map(command => (
+//             <tr key={command.id}>
+//               <td>{command.id}</td>
+//               <td>{command.name}</td>
+//               <td>{command.operator}</td>
+//               <td>{command.created}</td>
+//             </tr>
+//           ))}
+//         </tbody>
+//       </table>
+//       <div className="actions">
+//         <button className="add-command">add new command</button>
+//         <button className="see-logs">see logs</button>
+//       </div>
+//       <div className="footer">
+//         <button className="save-button">Save</button>
+//         <button className="cancel-button">Cancel</button>
+//       </div>
+//     </div>
+//   );
+// };
+
+// export default EditSchedulePage;
+
+import '../components/Scheduler.css'; // Import the CSS styles
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+import EditSchedulePage from "@/components/EditSchedules";
+import { useRouter } from "next/router";
+
+
+import { sendCommandSchedule } from "@/constants/api";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import React, { useEffect, useState } from "react";
+import ViewScheduleCard from "./ViewScheduleCard";
+import { color } from 'd3';
+import '../components/EditSchedules.css'; // Your stylesheet filename
+
+
+
+
+const Scheduler: React.FC = () => {
+  const queryClient = useQueryClient();
+  const router = useRouter();
+
+
+  // TODO: Dynamicall get satelliteId from somewhere
+  const satelliteId = "655acd63d122507055d3d2ea";
+
+  const adminUserId: string = "65a5e11fe0d601e0e8c4a385";
+
+  // admin
+  // const userId: string = "65a5e11fe0d601e0e8c4a385";
+
+  // operator
+  const userId: string = "65a8181f36ea10b4366e1dd9";
+
+  const scheduleId = "65a8182036ea10b4366e1de6";
+  const isAdmin = adminUserId === userId;
+
+  const [validCommands, setValidCommands] = useState([]);
+  const [currentSchedule, setCurrentSchedule] = useState<string[]>([]);
+
+  const fetchValidCommands = (satelliteId: string) => {
```

Comment:
if (and only if) you have time, put this into a utility file for reusability, since i wrote this already from a different file